horovod/spark/keras/remote.py

```diff
@@ -227,6 +227,12 @@ def train(serialized_model, train_rows, val_rows, avg_row_size):
                 reader_factory = make_batch_reader
                 is_batch_reader = True
 
+            # Call _setup again in process set module to point shared lib to tensorflow's module
+            # since the lib path might be overwritten in remote trainer.
+            _horovod.common.process_sets._setup(_horovod.tensorflow.mpi_ops._basics)
```

Comment:
This is an interesting one. Is this something we can fix a higher level of abstraction, like by calling `hvd.init()` again? I'm wondering if other non-estimator users might run into the same problem (like for `RayExecutor`).

Reply:
Yea, that's a good suggestion. I think we can hide it in init(), let me test it out.

Reply:
updated to move this logic to init()